src/components/MarkdownViewer.tsx

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+
+import React from 'react';
+import { Marked } from '@ts-stack/markdown';
+import DOMPurify from "dompurify";
+import {truncate} from "lodash";
+import { withTheme } from "@material-ui/core/styles";
+
+type Props = {
+  value?: string,
+  truncate: boolean,
+  theme: any,
+}
+
+type State = {
+  sanitizedValue: string,
+  sanitizedTruncatedValue: string,
+  shouldTruncate: boolean,
+  truncated: boolean,
+};
+
+const TRUNCATE_LENGTH = 100;
```

Comment:
It should truncate the warning message (the argument isn't provided to the component in Vue there and the default is to truncate), but not the check result message.

Your last screenshot above confuses me a bit because the UI wouldn't truncate the actual check result. Could it be that there's literally a "(truncated)" in your check result for some reason?

Reply:
It does literally contain `... (truncated)` which our backend does when a message is too long. Looks like the UI differences are simply differences in how the markdown rendering libraries work. Nothing to do.